src/widgets/DataTable/DataTableRow.js

```diff
@@ -42,187 +42,197 @@ import { formatStatus } from '../../utilities/index';
+                  width={width}
+                  isLastCell={isLastCell}
+                />
+              );
 
-          case 'reason':
-            return (
-              <DropDownCell
-                isDisabled={isFinalised}
-                dispatch={dispatch}
-                onPressAction={getAction(columnKey)}
-                rowKey={rowKey}
-                columnKey={columnKey}
-                value={rowData[columnKey]}
-                isLastCell={false}
-                width={width}
-                debug
-              />
-            );
+            case 'string': {
+              const value = rowData[columnKey];
+              const displayValue = columnKey === 'status' ? formatStatus(value) : value;
+              return (
+                <Cell
+                  key={columnKey}
+                  value={displayValue}
+                  width={width}
+                  viewStyle={cellContainer[cellAlignment]}
+                  textStyle={cellText[cellAlignment]}
+                  isLastCell={isLastCell}
+                />
+              );
+            }
 
-          default: {
-            const value = rowData[columnKey];
-            const displayValue = typeof value === 'number' ? Math.round(value) : value;
+            case 'numeric': {
+              return (
+                <Cell
+                  key={columnKey}
+                  value={Math.round(rowData[columnKey])}
+                  width={width}
+                  viewStyle={cellContainer[cellAlignment]}
+                  textStyle={cellText[cellAlignment]}
+                  isLastCell={isLastCell}
+                />
+              );
+            }
 
-            return (
-              <Cell
-                key={columnKey}
-                value={displayValue}
-                width={width}
-                viewStyle={cellContainer[cellAlignment]}
-                textStyle={cellText[cellAlignment]}
-                isLastCell={isLastCell}
-              />
-            );
-          }
-        }
-      });
-    }, [isFinalised, focusedColumnKey, rowState]);
+            case 'date':
+              return (
+                <Cell
+                  key={columnKey}
+                  value={rowData[columnKey] && rowData[columnKey].toDateString()}
+                  width={width}
+                  viewStyle={cellContainer[cellAlignment]}
+                  textStyle={cellText[cellAlignment]}
+                  isLastCell={isLastCell}
+                />
+              );
+
+            case 'icon':
+              return (
+                <TouchableCell
+                  key={columnKey}
+                  renderChildren={OpenModal}
+                  rowKey={rowKey}
+                  columnKey={columnKey}
+                  onPressAction={getAction(columnKey)}
+                  dispatch={dispatch}
+                  width={width}
+                  isLastCell={isLastCell}
+                  containerStyle={{
```

Comment:
Maybe should be in component `StyleSheet`?

Reply:
yeah